### DISCUSSION TITLE
Problem with launching swagger for Docker container

### DISCUSSION THREAD

**Discussion by PistonShot1** [dotnet/aspnetcore#55443] [Q&A]
Hi, I have build image and run a docker container locally. I mapped it to port 8000 to my local.
But when I make http request to ./swagger doesn't work , but I could direct to the endpoint 

i.e :
- `http://localhost:8000/WeatherForecast`  , works and sends response
- `http://localhost:8000/swagger/index.html` does not work

To setup the dockerfile and other I used `docker init`.

Here is the relevant , Dockerfile for your reference
```# syntax=docker/dockerfile:1

# Comments are provided throughout this file to help you get started.
# If you need more help, visit the Dockerfile reference guide at
# https://docs.docker.com/go/dockerfile-reference/

# Want to help us make this template better? Share your feedback here: https://forms.gle/ybq9Krt8jtBL3iCk7

################################################################################

# Learn about building .NET container images:
# https://github.com/dotnet/dotnet-docker/blob/main/samples/README.md

# Create a stage for building the application.
FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0-alpine AS build

COPY . /source

WORKDIR /source

# This is the architecture you’re building for, which is passed in by the builder.
# Placing it here allows the previous steps to be cached across architectures.
ARG TARGETARCH

# Build the application.
# Leverage a cache mount to /root/.nuget/packages so that subsequent builds don't have to re-download packages.
# If TARGETARCH is "amd64", replace it with "x64" - "x64" is .NET's canonical name for this and "amd64" doesn't
#   work in .NET 6.0.
RUN --mount=type=cache,id=nuget,target=/root/.nuget/packages \
    dotnet publish -a ${TARGETARCH/amd64/x64} --use-current-runtime --self-contained false -o /app

# If you need to enable globalization and time zones:
# https://github.com/dotnet/dotnet-docker/blob/main/samples/enable-globalization.md
################################################################################
# Create a new stage for running the application that contains the minimal
# runtime dependencies for the application. This often uses a different base
# image from the build stage where the necessary files are copied from the build
# stage.
#
# The example below uses an aspnet alpine image as the foundation for running the app.
# It will also use whatever happens to be the most recent version of that tag when you
# build your Dockerfile. If reproducability is important, consider using a more specific
# version (e.g., aspnet:7.0.10-alpine-3.18),
# or SHA (e.g., mcr.microsoft.com/dotnet/aspnet@sha256:f3d99f54d504a21d38e4cc2f13ff47d67235efeeb85c109d3d1ff1808b38d034).
FROM mcr.microsoft.com/dotnet/aspnet:8.0-alpine AS final
WORKDIR /app

# Copy everything needed to run the app from the "build" stage.
COPY --from=build /app .

# Switch to a non-privileged user (defined in the base image) that the app will run under.
# See https://docs.docker.com/go/dockerfile-user-best-practices/
# and https://github.com/dotnet/dotnet-docker/discussions/4764
USER $APP_UID

ENTRYPOINT ["dotnet", "pkt-epo-backend.dll"]
 ```
 
 I am quite new to .NET envrionment, perhaps it might be any configuration errors on my side, but running with `dotnet run` without container builds works fine for all endpoints including swagger endpoints. Do let me know if you suspect anything.
 Thanks in advance.

**Comment by PistonShot1** [comment]
Also , I forgot to mention that I am getting this warning when routing to any of the endpoints

``Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]``
``Failed to determine the https port for redirect``

I suspect it may be due to this warning as well , but I did look into it and modified still result in this warning. If you could point me to resolve this might resolve the issue above

**Reply by richlander** [reply]
That warning seems unrelated.

Is all the code for your app at the same location as the Docker context?

If you type `docker build -t app .`, the `.` is the location of the docker build context.

**Reply by PistonShot1** [reply]
I just realised , that my dockerfile was located within the project folder. I moved it outside and ran docker build and ran, still shows the same behaviour

**Reply by PistonShot1** [reply]
as for the warning , I think I need to expose 2 ports when i build the image as there are for http and https , which resolve that warning

**Reply by richlander** [reply]
Is this your issue?

https://stackoverflow.com/questions/67431299/how-to-access-asp-net-swagger-api-published-in-docker#comment119402339_67431653
